src/nix/store-import.cc

```diff
@@ -0,0 +1,88 @@
+#include "command.hh"
+#include "store-api.hh"
+
+using namespace nix;
+
+struct MixImportExport : virtual Args
+{
+    enum struct ArchiveFormat
+    {
+        Binary,
+    };
+
+    std::optional<ArchiveFormat> format = std::nullopt;
+    MixImportExport()
+    {
+        addFlag(
+            {.longName = "format",
+             .description = R"(
+                Format of the archive.
+                The only supported format is `binary`, which corresponds to the format used by [`nix-store --export`](@docroot@/command-ref/nix-store/export.md).
+            )",
+             .labels = {"format"},
+             .handler = {[&](std::string_view arg) {
+                 if (arg == "binary")
```

Comment:
Ideally the format identifier is usable out of context, because it's more recognizable and searchable that way.
E.g.

```suggestion
                 if (arg == "nix-store-object-stream-0")
```

thought: For version `1` it would be wise to add magic bytes. That would bring it up to NAR level, which has `nix-archive-1` for magic.

Reply:
We could lay groundwork to be able to say in the future something like `--format binary --format-version 1`

Reply:
Well, we can add that, but isn't that vastly overkill and confusing for something that we most likely won't want to evolve: We _might_ add new formats, which is why we settled on having this `--format` parameter, but if we ever add another one, that'll most likely be something totally different. And I really don't see anyone willing to pass `--format nix-store-object-stream-0`

Reply:
I can imagine this, but actually we can add the behavior I proposed exactly when needed. Therefore nevermind.